File: Interview Preparation Kit/Warm-up Challenges/Sock Merchant.js
function sockMerchant(n, ar) {
    if (n < 1 || n > 100) {
        return None;
    };
    var count = 0;
    var obj = {};
    for (let val of ar) {
        if (val < 1 || val > 100) {
            return None;
        }
        if (!obj[val]) {
            obj[val] = 1;
        } else {
            obj[val]++;
        }
    };
    for (const prop in obj) {
        count += Math.floor(obj[prop] / 2);
    };
    return count;

}

File: Interview Preparation Kit/Warm-up Challenges/Sock Merchant.py
def sockMerchant(n, ar):
    if n < 1 and n > 100:
        return None
    thisdict = {}
    count = 0
    for i in ar:
        if i < 1 and i >100:
            return None
        if str(i) not in thisdict:
            thisdict[str(i)] = 1
        else:
            thisdict[str(i)] += 1
    for j in thisdict:
        if thisdict[j] >= 2:
            count += thisdict[j]//2
    return count